bagherilab/ARCADE
# Update minibox filter function to support multiple tag separators

**Estimated time to review**: small

**Linked issue**: resolves #111 

**OLD Description of changes**:
- Changed check in MiniBox filter function from only returning keys with one tag separator to returning keys with any number of tag separators greater than 1
- Changed return value of filter function to be everything in the original key after the filtered code with tag separators replaced
- Amended the relevant unit test

**Final Description of changes**:
- Changed split in filter to only split on the first tag separator
- Changed return value of filter function to be everything in the original key after the filtered code with tag separators replaced
- Amended/added relevant unit tests

**How to verify changes**:
- Look at the test I amended. Running that test on main will fail.
- Add a distribution parameter to a proliferation module (ex. add this to the parameter.potts.xml file under the proliferation module parameters section: `<population.module module="proliferation" id="DIV_ROTATION_DISTRIBUTION" value="NORMAL(MU=0,SIGMA=0)" units="degrees" description="distribution that determines if plane of division is rotated" />`). Use a breakpoint to see that the distribution is constructed and stored in the Parameters object. Try doing this on main and see that all the other parameter distributions are constructed but not those in modules.
